Revert "Disable OSX builds"

peterschrammel opened the pull request:
This should be merged as soon as travis macos builds perform reasonably again.

Reverts diffblue/cbmc#1387

peterschrammel committed:
Revert "Disable OSX builds"

thk123 commented:
I suggest we restart the build every evening to see if it has got better (the last build took 5 hours so definitely not yet fixed...)

thk123 commented:
Ran this last night, took 2 hrs 28 min so I think that is still slow (albeit faster...) I will try again tonight.

thk123 commented:
Builds look reasonable again, but perhaps unsurprisingly looks like something broke while these were turned off. It appears to be to do with the new symbol table unit tests. Suggest poking @smowton

smowton committed:
Symbol-table writer: fix use of map key after erasure

This was revealed by re-enabling the OSX build, which evidently is quicker
to re-use the memory formerly occupied by the pair<key, value>!

smowton commented:
Could you rebase on https://github.com/diffblue/cbmc/pull/1520 to find out if that fixes it as expected?

smowton commented:
Never mind cherry-picked this into #1520

smowton commented:
Applied along with the corresponding fix in https://github.com/diffblue/cbmc/pull/1520
